src/nomad_simulations/model_system.py

```diff
@@ -967,6 +977,25 @@ def normalize(self, archive, logger) -> None:
         if is_not_representative(self, logger):
             return
 
+        # Traversing the System Hierarchy
+        # TODO consider moving to utils and passing generic set of functions to execute at each branch level
+        def set_branch_composition(system, subsystems, atom_labels):
```

Comment:
Why do we need to check `if self.composition_formula is None:`, I guess then the parser can explicitly set this quantity if we want?

Reply:
Also, the composition_formula is set for each branch and index of the tree individually, so it's much easier to set them within `traverse_system_recurs`....

Oh wait, I just realized that there may be some confusion or a problem here. I was sort of assuming that only branch_depth = 0 would be ever marked as representative. That's why I am traversing the entire tree to set the composition formula at each branch level. 

I could instead resolve the composition formula for all model systems, representative or not, then I don't need to have any recursive function, I just look at the children of the current branch. The difficulty there though is that for the bottom of the hierarchy I need to go back up the tree to find the list of atom labels :thinking:

Reply:
This is always the recurrent doubt I have (and that by good timing, @ladinesa just pointed out the contrary on another pr #52, he **does not** want to overwrite parsed data).

The condition of "is not None" is just because for me parsed data has priority, but maybe for you normalized one has it. We should do a decision so that we don't have conflicting parsing/normalization, and we should do it soon.

Reply:
Should we maybe add a warning that the composition formula was set by the user? Or maybe it's too annoying since it will get triggered a million times.

These warnings don't have any specific information about the instance, such that repeating warning is not really useful to 99% of users. @ladinesa is there any way to execute a warning such that duplicate warnings in the logger are ignored?